tests/app/data_model/test_questionnaire_store.py

```diff
@@ -1,185 +1,124 @@
-        store.set_metadata(non_serializable_metadata)
-        store.response_metadata = expected["RESPONSE_METADATA"]
-        store.answer_store = AnswerStore(expected["ANSWERS"])
-        store.progress_store = ProgressStore(expected["PROGRESS"])
-
-        # When / Then
-        self.assertRaises(TypeError, store.save)
-
-    def test_questionnaire_store_deletes(self):
-        # Given
-        expected = get_basic_input()
-        store = QuestionnaireStore(self.storage)
-        store.set_metadata(expected["METADATA"])
-        store.response_metadata = expected["RESPONSE_METADATA"]
-        store.answer_store = AnswerStore(expected["ANSWERS"])
-        store.progress_store = ProgressStore(expected["PROGRESS"])
-
-        # When
-        store.delete()  # See setUp - populates self.output_data
-
-        # Then
-        self.assertNotIn("a-test-section", store.progress_store)
-        self.assertEqual(store.metadata.copy(), {})
-        self.assertEqual(len(store.answer_store), 0)
-        self.assertEqual(store.response_metadata, {})
-
-    def test_questionnaire_store_raises_when_writing_to_metadata(self):
-        store = QuestionnaireStore(self.storage)
-
-        with self.assertRaises(TypeError):
-            store.metadata["no"] = "writing"
+def test_questionnaire_store_json_loads(questionnaire_store, basic_input):
+    # Given
+    questionnaire_store.input_data = json_dumps(basic_input)
+    # When
+    store = QuestionnaireStore(questionnaire_store.storage)
+    # Then
+    assert store.metadata.copy() == basic_input["METADATA"]
+    assert store.response_metadata == basic_input["RESPONSE_METADATA"]
+    assert store.answer_store == AnswerStore(basic_input["ANSWERS"])
+
+    expected_completed_block_ids = basic_input["PROGRESS"][0]["block_ids"][0]
+
+    assert (
+        len(store.progress_store.get_completed_block_ids("a-test-section", "abc123"))
+        == 1
+    )
+    assert (
+        store.progress_store.get_completed_block_ids("a-test-section", "abc123")[0]
+        == expected_completed_block_ids
+    )
+
+
+def test_questionnaire_store_ignores_extra_json(questionnaire_store, basic_input):
+    # Given
+    basic_input["NOT_A_LEGAL_TOP_LEVEL_KEY"] = "woop_woop_thats_the_sound_of_the_police"
+    questionnaire_store.input_data = json_dumps(basic_input)
+    # When
+    store = QuestionnaireStore(questionnaire_store.storage)
+    # Then
+    assert store.metadata.copy() == basic_input["METADATA"]
+    assert store.response_metadata == basic_input["RESPONSE_METADATA"]
+
+    assert store.answer_store == AnswerStore(basic_input["ANSWERS"])
+
+    expected_completed_block_ids = basic_input["PROGRESS"][0]["block_ids"][0]
+
+    assert (
+        len(store.progress_store.get_completed_block_ids("a-test-section", "abc123"))
+        == 1
+    )
+
+    assert (
+        store.progress_store.get_completed_block_ids("a-test-section", "abc123")[0]
+        == expected_completed_block_ids
+    )
```

Comment:
This is duplicated in `test_questionnaire_store_json_loads`. Could be a function call to avoid the dupe.